deeprank2/query.py

```diff
@@ -447,199 +452,52 @@ def build(self, feature_modules: List[ModuleType], include_hydrogens: bool = Fal
-            variant_amino_acid (:class:`AminoAcid`): The variant amino acid.
-            pssm_paths (Optional[Dict(str,str)], optional): The paths to the .PSSM files, per chain identifier. Defaults to None.
-            radius (float, optional): In Ångström, determines how many residues will be included in the graph. Defaults to 10.0.
-            distance_cutoff (Optional[float], optional): Max distance in Ångström between a pair of atoms to consider them as an external edge in the graph.
-                Defaults to 4.5.
-            targets (Optional[Dict(str,float)], optional): Named target values associated with this query. Defaults to None.
-            suppress_pssm_errors (bool, optional): Suppress error raised if .pssm files do not match .pdb files and throw warning instead.
-                Defaults to False.
-        """
-
-        self._pdb_path = pdb_path
-        self._pssm_paths = pssm_paths
-
-        model_id = os.path.splitext(os.path.basename(pdb_path))[0]
-
-        DeepRankQuery.__init__(self, model_id, targets, suppress_pssm_errors)
-
-        self._chain_id = chain_id
-        self._residue_number = residue_number
-        self._insertion_code = insertion_code
-        self._wildtype_amino_acid = wildtype_amino_acid
-        self._variant_amino_acid = variant_amino_acid
-
-        self._radius = radius
-
-        self._distance_cutoff = distance_cutoff
-
-    @property
-    def residue_id(self) -> str:
-        "String representation of the residue number and insertion code."
-
-        if self._insertion_code is not None:
-            return f"{self._residue_number}{self._insertion_code}"
-
-        return str(self._residue_number)
-
-    def get_query_id(self) -> str:
-        "Returns the string representing the complete query ID."
-        return f"atomic-graph:{self._chain_id}:{self.residue_id}:{self._wildtype_amino_acid.name}->{self._variant_amino_acid.name}:{self.model_id}"
-
-    def __eq__(self, other) -> bool:
-        return (
-            isinstance(self, type(other))
-            and self.model_id == other.model_id
-            and self._chain_id == other._chain_id
-            and self.residue_id == other.residue_id
-            and self._wildtype_amino_acid == other._wildtype_amino_acid
-            and self._variant_amino_acid == other._variant_amino_acid
-        )
+        if self.resolution == 'residue':
+            graph = build_residue_graph(residues, self.get_query_id(), self.distance_cutoff)
+        elif self.resolution == 'atomic':
+            residues.append(variant_residue)
+            atoms = set([])
+            for residue in residues:
+                if residue.amino_acid is not None:
+                    for atom in residue.atoms:
+                        atoms.add(atom)
+            atoms = list(atoms)
+            #TODO: why was this a set at first? I think each atom is unique anyway, given that it has a Residue property
```

Comment:
Your comment makes sense, but I guess there must be some reason why it could be that atoms are not unique. Otherwise indeed it could be a list from the beginning. What about asking Coos?

Reply:
I tested whether we can make them a list right away, and the tests now fail. I don't really understand why, but I think for the sake of wrapping this up I would just leave it as is (I had already removed the #TODO in a later commit)